### ISSUE TITLE
[DESIGN] Improve DX/UX of the public APIs

### ISSUE THREAD

**Issue by webknjaz** [ansible/pylibssh#91]
* [ ] Define the public APIs
* [ ] Separate internal APIs

Thoughts 💭
==========

1. It's not clear what are the usage scenarios of the lib (API-wise).
2. It is too low-level (mostly).
3. Most of the exposed classes implement managing resources and require calls to open/connect and close/disconnect. The idiomatic way of working with them is to implement context manager interfaces that guarantee that the managed resources are closed once unneeded, no matter what.
4. The usage sequence seems to require passing extra initialization params to the connect method. This is an antipattern: objects must be ready to use right away after the initialization. So things like credentials should be passed to init, not connect.

**Comment by ganeshrn** [comment]
Agreed the current API's are low level and we can add client class with context manager that handles session creation/deletion and command execution.
